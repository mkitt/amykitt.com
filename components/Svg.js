// @flow
import React from 'react'

type SvgProps = {
  children?: React.Element<*>,
  width?: number,
  height?: number,
  size?: number,
  viewBox: string,
}

const Svg = ({ children, size, width, height, viewBox, ...props }: SvgProps) => (
  <svg
    aria-hidden
    height={size || height}
    width={size || width}
    viewBox={viewBox}
    {...props}
  >
    {children}
  </svg>
)

Svg.defaultProps = {
  children: null,
  width: 32,
  height: 32,
  size: null,
}

export default Svg
