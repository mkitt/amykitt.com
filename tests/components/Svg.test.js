import React from 'react'
import renderer from 'react-test-renderer'
import Svg from '../../components/Svg'

describe('Svg', () => {
  it('shows the correct snapshot for the Svg component tree (width/height)', () => {
    const component = (
      <Svg width={24} height={24} viewBox="0 0 24 24">
        <g stroke="currentColor">
          <line x1="2" y1="2" x2="22" y2="22" />
          <line x1="22" y1="2" x2="2" y2="22" />
        </g>
      </Svg>
    )
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('shows the correct snapshot for the Svg component tree (size)', () => {
    const component = (
      <Svg size={24} viewBox="0 0 24 24">
        <g stroke="currentColor">
          <line x1="2" y1="2" x2="22" y2="22" />
          <line x1="22" y1="2" x2="2" y2="22" />
        </g>
      </Svg>
    )
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
