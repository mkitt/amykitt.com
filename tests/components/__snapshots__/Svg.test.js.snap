// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Svg shows the correct snapshot for the Svg component tree (size) 1`] = `
<svg
  aria-hidden={true}
  height={24}
  viewBox="0 0 24 24"
  width={24}
>
  <g
    stroke="currentColor"
  >
    <line
      x1="2"
      x2="22"
      y1="2"
      y2="22"
    />
    <line
      x1="22"
      x2="2"
      y1="2"
      y2="22"
    />
  </g>
</svg>
`;

exports[`Svg shows the correct snapshot for the Svg component tree (width/height) 1`] = `
<svg
  aria-hidden={true}
  height={24}
  viewBox="0 0 24 24"
  width={24}
>
  <g
    stroke="currentColor"
  >
    <line
      x1="2"
      x2="22"
      y1="2"
      y2="22"
    />
    <line
      x1="22"
      x2="2"
      y1="2"
      y2="22"
    />
  </g>
</svg>
`;
